lib/rspec/rails/matchers/active_job.rb

```diff
@@ -0,0 +1,71 @@
+require "active_job/base"
+
+module RSpec
+  module Rails
+    module Matchers
+      # Namespace for various implementations of ActiveJob features
+      #
+      # @api private
+      module ActiveJob
+        # @private
+        class HaveEnqueuedJobs < RSpec::Matchers::BuiltIn::BaseMatcher
+          def initialize(number, only = nil)
+            @number = number
+            @only   = only
+          end
+
+          def matches?(proc)
+            return false unless Proc === proc # TODO: better to raise error?
```

Comment:
Yes I think it'd be better to raise an error rather than silently failing, an argument error with a message would be best.